I'm getting a linking issue with your PR, though I can't find a reason why:

```
Linking CXX executable opendungeons
/usr/bin/ld: libangelscript.a(as_thread.cpp.o): undefined reference to symbol 'pthread_rwlock_wrlock@@GLIBC_2.2.5'
/lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
CMakeFiles/opendungeons.dir/build.make:2642: recipe for target 'opendungeons' failed
make[2]: *** [opendungeons] Error 1
CMakeFiles/Makefile2:95: recipe for target 'CMakeFiles/opendungeons.dir/all' failed
make[1]: *** [CMakeFiles/opendungeons.dir/all] Error 2
Makefile:137: recipe for target 'all' failed
make: *** [all] Error 2
```

I've tried to build again @oyvindln's branch, and it works fine. Will try to build development.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -286,6 +286,7 @@ set(OD_SOURCEFILES
 #### Find packages ###############
 ##################################
 
+find_package(Threads REQUIRED)
 find_package(OIS REQUIRED)
 find_package(OGRE REQUIRED)
 find_package(CEGUI REQUIRED)
@@ -396,6 +397,9 @@ add_executable(${PROJECT_BINARY_NAME} WIN32 ${OD_SOURCEFILES})
 ##################################
 #### Link libraries ##############
 ##################################
+
+# Add threads when linking AngelScript
+target_link_libraries (${AS_LIBRARY_NAME} ${CMAKE_THREAD_LIBS_INIT})
 
 # Link angelscript
 target_link_libraries(${PROJECT_BINARY_NAME} ${AS_LIBRARY_NAME})

--- a/README
+++ b/README
@@ -13,13 +13,16 @@ Data files :
 ------------
 
 config/          - Several game config files
+dist/            - Icons and linux desktop entry file.
 gui/             - CEGUI files + corresponding Gui images
+levels/          - Game levels
+licenses/        - License files used for game data and code
 materials/       - Materials (models texturing scripts and textures)
 models/          - Model files
 music/           - Music files
-sounds/          - Game Sounds
-levels/          - Game levels
+packs/           - Ogre files useful to run the starting pop-up
 scripts/         - Our AngelScript code files
+sounds/          - Game Sounds
 
 Code files:
 -----------
@@ -29,7 +32,6 @@ cmake/           - Helper files for CMake
  |- modules/     - Addon scripts for CMake to find dependencies
 dependencies/    - All external projects
  |- angelscript/ - AngelScript SDK, our scripting language
-documentation/   - Misc. files (incl. a Doxyfile and sketches)
 sources/         - All our own .cpp and .h files of the game
 .gitignore       - The files and folders that are ignored by git locally
 CMakeLists.txt   - CMake script for generating the Makefile and IDE projects

--- a/source/ODApplication.cpp
+++ b/source/ODApplication.cpp
@@ -51,8 +51,9 @@
 #include <fstream>
 
 ODApplication::ODApplication() :
-    mRoot(NULL),
-    mWindow(NULL)
+    mRoot(nullptr),
+    mWindow(nullptr),
+    mInitialized(false)
 {
     try
     {
@@ -123,6 +124,8 @@ ODApplication::ODApplication() :
         mRoot->addFrameListener(mFrameListener);
 
         mRoot->startRendering();
+
+        mInitialized = true;
     }
     catch(const Ogre::Exception& e)
     {
@@ -151,29 +154,34 @@ void ODApplication::cleanUp()
     LogManager* logManager = LogManager::getSingletonPtr();
     logManager->logMessage("Quitting main application...", Ogre::LML_CRITICAL);
 
-    logManager->logMessage("Stopping server...");
-    ODServer::getSingleton().stopServer();
-    logManager->logMessage("Disconnecting client...");
-    ODClient::getSingleton().disconnect();
+    if (mInitialized)
+    {
+        logManager->logMessage("Stopping server...");
+        ODServer::getSingleton().stopServer();
 
-    logManager->logMessage("Removing ODFrameListener from root...");
-    mRoot->removeFrameListener(mFrameListener);
-    logManager->logMessage("Deleting ODFrameListener...");
-    delete mFrameListener;
-    logManager->logMessage("Deleting Ogre::OverlaySystem...");
-    delete mOverlaySystem;
-    logManager->logMessage("Deleting MusicPlayer...");
-    delete MusicPlayer::getSingletonPtr();
-    logManager->logMessage("Deleting TextRenderer...");
-    delete TextRenderer::getSingletonPtr();
-    logManager->logMessage("Deleting Gui...");
-    delete Gui::getSingletonPtr();
-    logManager->logMessage("Deleting SoundEffectsManager...");
-    delete SoundEffectsManager::getSingletonPtr();
-    logManager->logMessage("Deleting ODServer...");
-    delete ODServer::getSingletonPtr();
-    logManager->logMessage("Deleting ODClient...");
-    delete ODClient::getSingletonPtr();
+        logManager->logMessage("Disconnecting client...");
+        ODClient::getSingleton().disconnect();
+
+        logManager->logMessage("Removing ODFrameListener from root...");
+        mRoot->removeFrameListener(mFrameListener);
+        logManager->logMessage("Deleting ODFrameListener...");
+        delete mFrameListener;
+        logManager->logMessage("Deleting Ogre::OverlaySystem...");
+        delete mOverlaySystem;
+        logManager->logMessage("Deleting MusicPlayer...");
+        delete MusicPlayer::getSingletonPtr();
+        logManager->logMessage("Deleting TextRenderer...");
+        delete TextRenderer::getSingletonPtr();
+        logManager->logMessage("Deleting Gui...");
+        delete Gui::getSingletonPtr();
+        logManager->logMessage("Deleting SoundEffectsManager...");
+        delete SoundEffectsManager::getSingletonPtr();
+        logManager->logMessage("Deleting ODServer...");
+        delete ODServer::getSingletonPtr();
+        logManager->logMessage("Deleting ODClient...");
+        delete ODClient::getSingletonPtr();
+    }
+
     logManager->logMessage("Deleting ConfigManager...");
     delete ConfigManager::getSingletonPtr();
     logManager->logMessage("Deleting ResourceManager...");
@@ -181,6 +189,8 @@ void ODApplication::cleanUp()
     logManager->logMessage("Deleting LogManager and Ogre::Root...");
     delete LogManager::getSingletonPtr();
     delete mRoot;
+
+    mInitialized = false;
 }
 
 //TODO: find some better places for some of these

--- a/source/ODApplication.h
+++ b/source/ODApplication.h
@@ -63,6 +63,10 @@ private:
     //! \brief by both the RenderManager and the TextRenderer.
     Ogre::OverlaySystem* mOverlaySystem;
 
+    //! \brief Keep in mind whether the game has been properly initialized.
+    //! Used when leaving the application in case someone cancelled the Ogre config pop-up.
+    bool mInitialized;
+
     //! \brief Delete the various singleton objects and clean up other stuff
     void cleanUp();
 };

